File: PharmaDicBackEnd/PharmaDicBackEnd.ApiService/Models/MedicineCategory.cs
using System;
using System.Collections.Generic;

namespace PharmaDicBackEnd.ApiService.Models;

public partial class MedicineCategory
{
    public int CategoryId { get; set; }

    public string CategoryName { get; set; } = null!;

    public string? Description { get; set; }

    public virtual ICollection<Medicine> Medicines { get; set; } = new List<Medicine>();
}

File: PharmaDicMobile/app/src/main/java/com/example/pharmadicmobile/models/MedicineCategory.java
package com.example.pharmadicmobile.models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class MedicineCategory implements Serializable {
    private int categoryId;
    private String categoryName;
    private String description;
    private List<Medicine> medicines = new ArrayList<>();

    public MedicineCategory() {}

    public int getCategoryId() { return categoryId; }
    public void setCategoryId(int categoryId) { this.categoryId = categoryId; }

    public String getCategoryName() { return categoryName; }
    public void setCategoryName(String categoryName) { this.categoryName = categoryName; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public List<Medicine> getMedicines() { return medicines; }
    public void setMedicines(List<Medicine> medicines) { this.medicines = medicines; }
}
